### PR TITLE
Added docker-build-push workflow and added to README

### PR DESCRIPTION
**Pull request by jaxels10**
**Description of your changes:**
I have added a workflow that finds all our Dockerfiles, tags the image with the helm chart version, build and pushes it. 
Furthermore, I have added to the README how to build dockerfiles. 
EDIT: I have noticed that the loadbalancer steps I added to the lint-and-test workflow do not check if any charts are changed and therefore tries to create a loadbalancer on a kind cluster that was not created. This has been included in this PR. The checks on this PR will probably fail due to this not being on the main branch yet. 

Checklist:

* [ ] I have bumped the chart version
* [x] I have documented my changes if this is needed
* [x] I have described my changes in the "description of your changes" field
* [ ] I have tested this change on a running cluster and the Argocd dashboard shows healthy and synced
* [ ] I have created the necessary tests in the chart, if they are possible/necessary
* [ ] I have squashed commits if necessary

### CODE REVIEW

**Comment by klausenbusk** [comment]
Should this PR also move the existing Dockerfiles to `chart/<foo>/container/Dockerfile`?

File: .github/workflows/docker-build-push.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+          fetch-depth: '0'
+      
+      - name: Check file existence
+        id: check_files
+        uses: andstor/file-existence-action@v1
+        with:
+          files: "charts/${{ matrix.path }}/container/Dockerfile"
+
+      - name: Log in to the Container registry
+        if: steps.check_files.outputs.files_exists == 'true'
```
Comment (skovbakke) [inline .github/workflows/docker-build-push.yaml:35]: I think it is possible to move the "Check file existence" step into its own job, have that job output a boolean, and have the second job check if it should run or not. That way we only need to have this if statement in one place. 
eg:
```yaml
jobs:
  job1: 
    outputs: 
      output1: ${{ steps.check_files.outputs.files_exists }}
    steps:
       - name: check file existence
         ...
  job2:
    needs: job1
    if ${{ needs.job1.outputs.output1 }}
    steps:
      <All the other steps> 
      ...
```

**Review by skovbakke — CHANGES_REQUESTED** [review]
Small suggestion 😊

File: .github/workflows/docker-build-push.yaml (hunk 1 above)
Comment (jaxels10) [reply .github/workflows/docker-build-push.yaml:35]: That does look a lot better yes, but will this work with the way the matrix is iterating the paths? The check file existence step takes a path variable from the matrix. If I move this step out, I will have to also create a matrix for the first job, but I do not think it is possible to call a job from another job.

Comment (skovbakke) [reply .github/workflows/docker-build-push.yaml:35]: I see. I did not consider the matrix. Somehow each entry in the matrix creates its own job, and I could see the code getting more messy if we try to implement my suggestion.

**Comment by jaxels10** [comment]
> Should this PR also move the existing Dockerfiles to `chart/<foo>/container/Dockerfile`?

I am unsure what your thoughts are with this? The Dockerfiles should already be located at charts/<foo>/container/Dockerfile

**Comment by klausenbusk** [comment]
> The Dockerfiles should already be located at charts//container/Dockerfile

Doesn't appear to be the case:
* https://github.com/distributed-technologies/helm-charts/blob/main/charts/git-server/Dockerfile
* https://github.com/distributed-technologies/helm-charts/blob/main/charts/ntp/Dockerfile

**Comment by jaxels10** [comment]
> > The Dockerfiles should already be located at charts//container/Dockerfile
> 
> Doesn't appear to be the case:
> 
> * https://github.com/distributed-technologies/helm-charts/blob/main/charts/git-server/Dockerfile
> * https://github.com/distributed-technologies/helm-charts/blob/main/charts/ntp/Dockerfile

Ahh okay now I understand. 

I have just moved the docker relevant files for both the NTP and git-server chart into the container/ folder. I have not touched Kafka, since jacob said I should not.

**Review by klausenbusk — APPROVED** [review]
LGTM 🎉

Let's add support for building images for feature branches in another PR.

**Pull request merged**